modules/renter/contractor/editor.go

```diff
@@ -181,42 +203,68 @@ func (he *hostEditor) Delete(root crypto.Hash) error {
 	if index == -1 {
 		return errors.New("no record of that sector root")
 	}
-	tree := crypto.NewCachedTree(sectorHeight) // NOTE: height is not strictly necessary here
-	for _, h := range newRoots {
-		tree.Push(h)
-	}
-	merkleRoot := tree.Root()
+	merkleRoot := cachedMerkleRoot(newRoots)
 
-	// initiate revision
-	if err := startRevision(he.conn, he.host, he.contractor.hdb); err != nil {
-		return err
-	}
-
-	// send 'delete' action
-	err := encoding.WriteObject(he.conn, []modules.RevisionAction{{
+	// create the action and accompanying revision
+	actions := []modules.RevisionAction{{
 		Type:        modules.ActionDelete,
 		SectorIndex: uint64(index),
-	}})
-	if err != nil {
-		return err
+	}}
+	rev := newRevision(he.contract.LastRevision, merkleRoot, uint64(len(newRoots)), sectorPrice, sectorCollateral)
+
+	// run the revision iteration
+	return he.runRevisionIteration(actions, rev, newRoots, height)
+}
```

Comment:
interesting that you would have each action be a separate function, that's going to make batching more difficult later